src/collect/output.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    fs::{File, OpenOptions},
+    io::{BufWriter, Write},
+    time::Duration,
+};
+
+use anyhow::Result;
+use log::error;
+
+use crate::core::events::bpf::BpfEvents;
+
+/// Controls the output of events for the collector cmd.
+pub(super) struct Output<'a> {
+    /// Reference to the BPF events, mainly used to retrieve events.
+    bpf_events: &'a BpfEvents,
+    /// Should we output messages to stdout?
+    quiet: bool,
+    /// Writer to a file to store events for later processing, if any.
+    writer: Option<BufWriter<File>>,
+    /// Output file used in the above writer.
+    output_file: Option<File>,
+}
```

Comment:
This output struct assumes the only way of printing events is in json (which is not very readable tbh).
If we aim at supporting various ways of printing data even on collection phase, how about we define a simple trait such as:
```rust
trait Processor {
    process_one(e: Event) -> Result<()>,
    close() -> Result<()>
}
```
Then this `FileOutput` would implement the trait (and only output to a file) but allow us to combine more than one simultaneous processor easily, some that come to mind at first `JsonPrint` (existing functionality), `PrettierThanJsonPrint`, `StatsPrint`,  `Spinner`, `SendToSock`, etc.

Reply:
It's a good idea.
My only concern is that there might be `Processors` and `Write` that don't make much sense together, like a textual-like table and a UDP socket. On the other hand, could there be `Processor` implementation that need a bit more than just a `Write`? e.g: a TUI interface?

If so, we could have  something like:
```rust
trait Processor {
     process_one(e: Event) -> Result<()>,
}

trait Formatter {...} 
trait Write {...}

struct FormatAndWrite<F: Formatter,  W: Write>  {
     formatter: &F,
     writers: Vec<W>,
}
impl<F, W>Processor for  FormatAndWrite<F, W> {
    process_one(e: Event) -> Result<()> {
            let bytes = self.formatter.format(e);
            for w in self.writers { 
                   w.write(bytes);
            }
    }
}
```

That would leave space for all kind of processing. But maybe this is over-engineering something that's way simpler right now.